src/jvm/main/org/jetbrains/kotlinx/lincheck/Utils.kt

```diff
@@ -81,7 +82,15 @@ private fun executeValidationFunction(instance: Any, validationFunction: Method)
     try {
         m.invoke(instance)
     } catch (e: Throwable) {
-        return e.cause
+        // cut off lincheck-related and reflection-related stacktrace elements
+        val validationException = e.cause ?: return e
+        val wrapperExceptionStackTraceLength = e.stackTrace.size
+
+        if (e is InvocationTargetException) {
```

Comment:
Can it be not an `InvocationTargetException`?

Reply:
Please add a comment explaining when another exception type can be thrown, clarifying why simply returning it is correct. Maybe, we need to throw an internal exception instead.

Reply:
Done